### PR TITLE
Add request for documentation.

### LINKED ISSUE #1919: Make it mandatory to document modified or newly created functions/structs in PRs
### Is your feature request related to a problem? Please describe.

As a newbie with rizin and as not so experienced programmer it was/is relatively hard to understand the code base.

E.g. when I try to fix a bug, quite a lot of time goes into understanding the path of execution in order to pinpoint the function which _could_ be the problem.
 
Mostly because quite a lot of functions have no short description what they are doing.
This is valid even more so for some single letter variables, parameters and struct members.

### Describe the solution you'd like

It is way to much work to document everything at once. Hence:

Add another checkbox in the PR template which asks specifically for function/struct documentation. Something like:
 _"I have documented or updated the documentation of every function and struct this PR changes."_

This request to document also includes existing functions. If they were touched the PR owner has likely understood them and can therefore document them.

### Additional context

I can't assess yet how much of this problem is because of my inexperience reading code and/or laziness to invest time understanding it.
Also I guess there are also disadvantages documenting each and every function/struct? Too much information for too little gain etc.?
What do you think?

### PR DESCRIPTION
**Pull request by Rot127**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [ ] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Basically adds the request to document all code.

Newly added code, as well as old code the creator has changed.

The idea is to improve (especially) old code. Every PR which touches old and/or undocumented code, can be used to add or update the documentation of it.
Especially since the PR creator has probably understood the code she has edited. 

Relevant issue: https://github.com/rizinorg/rizin/issues/1919

**Test plan**

Not needed.

**Closing issues**

closes https://github.com/rizinorg/rizin/issues/1919

### CODE REVIEW

**Review by ret2libc — APPROVED** [review]
I like it. cc @XVilka @wargio @thestr4ng3r @ITAYC0HEN @kazarmy

File: DEVELOPERS.md [hunk 1]
```diff
@@ -30,6 +30,15 @@ for more info. Example usage can be found [here](http://www.doxygen.nl/manual/do
 static int findMinMax(RzList *maps, ut64 *min, ut64 *max, int skip, int width);
 ```
 
+In order to improve the documentation and help newcomers, documenting code is mandatory.
+
+You should add or update the documentation of:
+- code written by you.
+- existing Rizin code you changed.
```
Comment (XVilka) [inline DEVELOPERS.md:37]: Maybe add an exception for trivial changes, since often those could be done without understanding what the modified function does.

**Review by XVilka — CHANGES_REQUESTED** [review]

File: DEVELOPERS.md (hunk 1 above)
Comment (Rot127) [reply DEVELOPERS.md:37]: Adding an exception besides the example?
Or might it be enough to add another example here:

https://github.com/rizinorg/rizin/blob/0db022224578652374cc9127131f9ca888f21fc5/DEVELOPERS.md?plain=1#L39-L40
Something like:
"Changes are trivial and don't need documentation."
"Getter and setter are trivial and don't need documentation."

Comment (Rot127) [reply DEVELOPERS.md:37]: Never mind. The exception clause is more inclusive.

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Review by officialcjunior — APPROVED** [review]
